### ISSUE TITLE
"Requested operation has changed during the UI authentication session." shouldn't be sent if only the query string changes

### ISSUE THREAD

**Issue by babolivier** [matrix-org/synapse#7722]
https://github.com/matrix-org/synapse/pull/7483 changed the condition under which the "Requested operation has changed during the UI authentication session." error raised so that it's only raised if the (method, uri) changes during UIA.

This means that if e.g. I initiate a registration with `POST /register` but ends it with `POST /register?kind=user`, which is the same operation, the registration fails.

Ideally this condition should be changed to omit the query string from the uri, but we should first check if that doesn't introduce any issue (e.g. if the UIA operation is passed in the query string in some cases - which I don't think it is, but we should probably make sure of that).

**Comment by clokep** [comment]
We also need to make sure that things like device IDs, etc. are not part of the query string, but part of the path.

**Comment by clokep** [comment]
@babolivier Any idea how much of an impact this might have?

**Comment by babolivier** [comment]
I don't think this is impacting many clients - the main reason I found out about that is because the web version of Tchap has this issue.
